terraform/account-wide-infrastructure/modules/glue/s3.tf

```diff
@@ -0,0 +1,146 @@
+  bucket = "source-data-bucket"
+}
+
+resource "aws_s3_bucket_policy" "source-data-bucket" {
+  bucket = "source-data-bucket"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Id      = "source-data-bucket-policy"
+    Statement = [
+      {
+        Sid    = "HTTPSOnly"
+        Effect = "Deny"
+        Principal = {
+          "AWS" : "*"
+        }
+        Action = "s3:*"
+        Resource = [
+          aws_s3_bucket.source-data-bucket.arn,
+          "${aws_s3_bucket.source-data-bucket.arn}/*",
+        ]
+        Condition = {
+          Bool = {
+            "aws:SecureTransport" = "false"
+          }
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_s3_bucket_public_access_block" "source-data-bucket-public-access-block" {
+  bucket = aws_s3_bucket.source-data-bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+
+# S3 Bucket for Processed Data
+resource "aws_s3_bucket" "target-data-bucket" {
+  bucket = "target-data-bucket"
+}
+
+resource "aws_s3_bucket_policy" "target-data-bucket" {
+  bucket = "target-data-bucket"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Id      = "target-data-bucket-policy"
+    Statement = [
+      {
+        Sid    = "HTTPSOnly"
+        Effect = "Deny"
+        Principal = {
+          "AWS" : "*"
+        }
+        Action = "s3:*"
+        Resource = [
+          aws_s3_bucket.target-data-bucket.arn,
+          "${aws_s3_bucket.target-data-bucket.arn}/*",
+        ]
+        Condition = {
+          Bool = {
+            "aws:SecureTransport" = "false"
+          }
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_s3_bucket_public_access_block" "target-data-bucket-public-access-block" {
+  bucket = aws_s3_bucket.target-data-bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+# S3 Bucket for Code
+resource "aws_s3_bucket" "code-bucket" {
+  bucket = "code-bucket"
+}
+
+resource "aws_s3_bucket_policy" "code-bucket" {
+  bucket = "code-bucket"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Id      = "code-bucket-policy"
+    Statement = [
+      {
+        Sid    = "HTTPSOnly"
+        Effect = "Deny"
+        Principal = {
+          "AWS" : "*"
+        }
+        Action = "s3:*"
+        Resource = [
+          aws_s3_bucket.code-bucket.arn,
+          "${aws_s3_bucket.code-bucket.arn}/*",
+        ]
+        Condition = {
+          Bool = {
+            "aws:SecureTransport" = "false"
+          }
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_s3_bucket_public_access_block" "code-bucket-public-access-block" {
+  bucket = aws_s3_bucket.code-bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_object" "code-data-object" {
```

Comment:
It's also complaining when running `terraform plan` about a few other s3 bucket releated things too:
![Screenshot from 2024-12-13 13-57-57](https://github.com/user-attachments/assets/87d19f13-6a09-4c21-8e56-c06d72014948)